uk.co.compucorp.civicrm.hrcomments/api/v3/Comment.php

```diff
@@ -31,16 +31,60 @@ function civicrm_api3_comment_create($params) {
  * @throws API_Exception
  */
 function civicrm_api3_comment_delete($params) {
-  return _civicrm_api3_basic_delete(_civicrm_api3_get_BAO(__FUNCTION__), $params);
+  return _civicrm_api3_comment_soft_delete(_civicrm_api3_get_BAO(__FUNCTION__), $params);
 }
 
 /**
  * Comment.get API
+ * This API returns comments that are not deleted
+ * i.e comments with is_deleted flag false.
  *
  * @param array $params
+ *
  * @return array API result descriptor
+ *
  * @throws API_Exception
  */
 function civicrm_api3_comment_get($params) {
-  return _civicrm_api3_basic_get(_civicrm_api3_get_BAO(__FUNCTION__), $params);
+  $results =  _civicrm_api3_basic_get(_civicrm_api3_get_BAO(__FUNCTION__), $params);
+
+  if ($results['count'] > 0){
+    $nonDeleted = array_filter($results['values'], function($item) {
+      return !$item['is_deleted'];
+    });
+    $results['values'] = array_values($nonDeleted);
+  }
+
```

Comment:
mmmm... there are a few problems with this implementation:
- It will mess with `$results['count']`, because you removing items from returned values but the count number will still be the original number of items returned
- It will mess with pagination (limit and offset). Example: say you have 6 comments and only the first one of them is deleted. Then you use `limit` to only get the first 5, which will include the deleted one. The code will remove the deleted comment and will actually return only 4 comments, even though, in the database, we have 5 non deleted comments.

In order to avoid these problems, you'll need to play with the `Civi\API\SelectQuery` class, which is what is used internally by the `get` API. Take a look at [API/Query/LeaveRequestSelect.php](https://github.com/civicrm/civihr/blob/la-milestone-3/uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/API/Query/LeaveRequestSelect.php) and [api/v3/LeaveRequest.php](https://github.com/civicrm/civihr/blob/la-milestone-3/uk.co.compucorp.civicrm.hrleaveandabsences/api/v3/LeaveRequest.php#L108-L111) to see how something similar was done for the LeaveRequest API